moabb/paradigms/ssvep.py

```diff
@@ -0,0 +1,185 @@
+"""Steady-State Visually Evoked Paradigms"""
+
+import mne
+import numpy as np
+import pandas as pd
+import logging
+
+from moabb.paradigms.base import BaseParadigm
+from moabb.datasets import utils
+from moabb.datasets.fake import FakeDataset
+
+log = logging.getLogger()
+
+
+class BaseSSVEP(BaseParadigm):
+    """Base SSVEP Paradigm
+
+    Parameters
+    ----------
+    filters: list of list | None (default None)
+        Bank of bandpass filter to apply. If None, bandpass set around freqs
+        with [f_n-0.5, f_n+0.5] and a global one [1, 45]
+    freqs: list of str | None (default None)
+        List of stimulation frequencies (float). If None, use all stimulus
+        found in the dataset.
+    n_classes: int,
+        Number of classes each dataset must have.
+    tmin: float (default 0.0)
+        Start time (in second) of the epoch, relative to the dataset specific
+        task interval e.g. tmin = 1 would mean the epoch will start 1 second
+        after the begining of the task as defined by the dataset.
+    tmax: float | None, (default None)
+        End time (in second) of the epoch, relative to the begining of the
+        dataset specific task interval. tmax = 5 would mean the epoch will end
+        5 second after the begining of the task as defined in the dataset. If
+        None, use the dataset value.
+    channels: list of str | None (default None)
+        List of channel to select. If None, use all EEG channels available in
+        the dataset.
+    resample: float | None (default None)
+        If not None, resample the eeg data with the sampling rate provided.
+    """
+
+    def __init__(self, filters=None, freqs=None, tmin=0.0, tmax=None,
+                 channels=None, n_classes=None, resample=None):
+        super().__init__()
+        self.filters = filters
+        self.freqs = freqs
+        self.n_classes = n_classes
+        self.channels = channels
+        self.resample = resample
+
+        if (tmax is not None):
+            if tmin >= tmax:
+                raise(ValueError("tmax must be greater than tmin"))
+
+        self.tmin = tmin
+        self.tmax = tmax
+
+    def is_valid(self, dataset):
+        ret = True
+        if not (dataset.paradigm == 'ssvep'):
+            ret = False
+        if self.freqs is None and self.n_classes is None:
+            return ret
+
+        if self.freqs is None and self.n_classes is not None:
+            if not len(dataset.event_id) >= self.n_classes:
+                ret = False
+        else:
+            overlap = len(set(self.freqs) & set(dataset.event_id.keys()))
+            if not overlap >= self.n_classes:
+                ret = False
+
+        # we should verify list of channels, somehow
```

Comment:
code to do that exists in dataset.utils, it's done in the dataset_search function